source/utilities.cc

```diff
@@ -1643,9 +1643,17 @@ namespace aspect
         }
       while (in >> temp_data);
 
+      AssertThrow(in.eof(),
+                  ExcMessage (std::string("While reading the data file '" + filename + "' the ascii data "
+                                          "plugin has encountered an error before the end of the file. "
+                                          "Please check for malformed data values (e.g. NaN) or superfluous "
+                                          "lines at the end of the data file.")));
 
       AssertThrow(field_index == (components + dim) * data_table.n_elements(),
-                  ExcMessage (std::string("Number of read in points does not match number of expected points. File corrupted?")));
+                  ExcMessage (std::string("While reading the data file '" + filename + "' the ascii data "
```

Comment:
I think @jdannberg  wants this assert() condition (which appears in the output) to be easier to understand (right?). Maybe we can do 
```suggestion
      const n_total_values = (components + dim) * data_table.n_elements();
      AssertThrow(field_index == n_total_values,
```
and ``... expected number of floating point values based on the dimension, the number of components, and the number of lines given by POINTS in the header of the file``.